### ISSUE TITLE
Cannot access connector_server from yubihsm-shell and connector_server_cli from auth_key by running tmkms yubihsm-server

### ISSUE THREAD

**Issue by juggernaut09** [iqlusioninc/tmkms#402]
I cannot access the `connector_server` running at `tcp://localhost:12345`  

When I `curl` the URL the results are working fine:
```
root@tmkms:~# curl http://localhost:12345/connector/status
status
OK
serial
*
version
0.38.0
pid
11319
address
127.0.0.1
port
12345
```
The `yubihsm-shell` logs says:

```
root@tmkms:~# yubihsm-shell
Using default connector URL: http://127.0.0.1:12345
yubihsm> debug all
Debug messages enabled
yubihsm> connect
[LIB - INF 05:34:31.405561] yubihsm.c:4127 (yh_init_connector): Loading http backend
[LIB - INF 05:34:31.418125] yubihsm_curl.c:85 (backend_create): backend_create
[LIB - INF 05:34:31.418159] yubihsm_curl.c:90 (backend_connect): backend_connect
[LIB - INF 05:34:31.418166] yubihsm_curl.c:97 (backend_connect): Trying to connect to http://127.0.0.1:12345/connector/status
[LIB - INF 05:34:31.418719] lib_util.c:139 (parse_status_data): response from connector
[LIB - INF 05:34:31.418755] lib_util.c:140 (parse_status_data): has device: no
[LIB - INF 05:34:31.418778] lib_util.c:141 (parse_status_data): version: 0.0.0
[LIB - INF 05:34:31.418804] lib_util.c:143 (parse_status_data): pid: 0
[LIB - INF 05:34:31.418830] lib_util.c:144 (parse_status_data): address: 
[LIB - INF 05:34:31.418856] lib_util.c:145 (parse_status_data): port: 0
[LIB - ERR 05:34:31.418883] yubihsm_curl.c:138 (backend_connect): Failure when connecting: Connector has no device
[LIB - ERR 05:34:31.418907] yubihsm.c:4166 (yh_connect): Failed when connecting: Unable to find a suitable connector
Failed connecting 'http://127.0.0.1:12345': Unable to find a suitable connector
[LIB - INF 05:34:31.418952] yubihsm_curl.c:150 (backend_disconnect): backend_disconnect
[LIB - INF 05:34:31.419167] yubihsm_curl.c:212 (backend_cleanup): backend_cleanup
yubihsm> 
```

This is the `tmkms.toml` provider configuration I am using:
```
[[providers.yubihsm]]
adapter = { type = "usb" }
auth = { key = 1, password_file = "/root/.tmkms/secrets/yubihsm-password.txt" }
#serial_number = "0123456789" # serial number of a specific YubiHSM to connect to (optional)
keys = [
    { key = 5, type = "consensus", chain_ids = ["regen-redwood-1"] },
    { key = 6, type = "consensus", chain_ids = ["flixnet-2"] }
]
connector_server = { laddr = "tcp://127.0.0.1:12345", cli = { auth_key = 2 } }
```

- I am able to access the `connector_server_cli` using `{ auth_key = 2 }` via `tmkms yubihsm keys` subcommands.
- And tmkms server is perfectly signing two validators transactions.
- Only problem is not able to access the `connector_server` via `yubihsm-shell`

**Comment by tony-iqlusion** [comment]
This is likely caused by an incompatibility between the latest `yubihsm-shell` and the connector server built into TMKMS.

One thing that would be helpful for debugging is if you could run the normal `yubihsm-connector` service and post the response from the `/connector/status` endpoint.

Otherwise, I will try to find some time to debug this, but it probably won't be soon. If you can get me the `/connector/status` endpoint response, I can probably look at it sooner.

**Comment by juggernaut09** [comment]
- Hi, @tony-iqlusion. Thanks for replying We have started using `tmkms` for signing Mainnet validators a few days back.
- However, We had to remove the `connector_server` from the `yubihsm_config` and run the `tmkms` server because of critical urgency.
- We are able to perform the `signing operations` well. But cannot access the `yubihsm-shell` separately.
- Now we don't have the option to stop the running `tmkms` server and run the `yubihhsm-connector` server.
- If we run the `yubihsm-connector` server. It says `USB is busy` because of the existing USB allotment with the`tmkms` server.
- As far as I remember. `/connector/status` of `yubihsm-connector` server resulted same as the below-mentioned status.
```
status
OK
serial
*
version
0.38.0
pid
11319
address
127.0.0.1
port
12345
```
Regards.
